cfgov/teachers_digital_platform/views.py

```diff
@@ -188,14 +189,17 @@ def view_results(request: HttpRequest):
     if request.method != 'GET':
         return HttpResponse(status=404)
 
+    if 'r' not in request.GET:
+        return HttpResponseRedirect(_unexpected_error_uri)
```

Comment:
Any time we access a request parameter it should really be using [a Django form](https://docs.djangoproject.com/en/2.2/topics/forms/). We get this kind of required field validation for free in a paradigmatic way, along with any particular built-in sanitization we might need for a particular parameter type.

Reply:
We'll definitely need a unit test around expected inputs for `r` here, regardless.

Reply:
@willbarton We have cypress tests for it, but I can look into the form route too. Using a form would require more custom JS to build the URL because the user that completes the survey never visits the URL but needs to have it.

Reply:
A Django form doesn't have to imply anything on the front-end. It can simply handle the input validation and sanitization, and is the preferred approach to do so. An example is [where we use a form class to handle the autocomplete field in Ask CFPB](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/ask_cfpb/forms.py#L13-L17) and [check its validity in the view](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/ask_cfpb/views.py#L147-L151) but do not use it for rendering.

The other thing is that Cypress is used for functional testing. It's not how we unit test our Python, nor does it indicate coverage of our Python code.

Reply:
Yeah, definitely more python units are coming. I'll give the form a shot.